src/main/kotlin/io/gitlab/arturbosch/detekt/idea/action/AddToBaselineAction.kt

```diff
@@ -0,0 +1,45 @@
+package io.gitlab.arturbosch.detekt.idea.action
+
+import com.intellij.codeInsight.daemon.DaemonCodeAnalyzer
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.components.service
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import io.github.detekt.tooling.api.BaselineProvider
+import io.gitlab.arturbosch.detekt.api.Finding
+import io.gitlab.arturbosch.detekt.idea.DETEKT
+import io.gitlab.arturbosch.detekt.idea.DetektBundle
+import io.gitlab.arturbosch.detekt.idea.config.DetektPluginSettings
+import kotlin.io.path.Path
+
+class AddToBaselineAction(private val finding: Finding) : IntentionAction {
+
+    override fun startInWriteAction(): Boolean = true
+
+    override fun getText(): String = DetektBundle.message("detekt.actions.addToBaseline", finding.id)
+
+    override fun getFamilyName(): String = DETEKT
+
+    override fun isAvailable(project: Project, editor: Editor, file: PsiFile): Boolean {
+        val id = finding.compactWithSignature()
+        return id.trim().isNotEmpty() && project.isBaselineDefined()
+    }
+
+    private fun Project.isBaselineDefined(): Boolean = service<DetektPluginSettings>().baselinePath.isNotEmpty()
+
+    override fun invoke(project: Project, editor: Editor, file: PsiFile) {
+        val settings = project.service<DetektPluginSettings>()
+        if (project.isBaselineDefined()) {
+            val provider = BaselineProvider.load()
+            val baselinePath = Path(settings.baselinePath)
```

Comment:
Each Gradle module can have its own Detekt baseline, right? I mean right now, all IDEA knows is what's set manually in Settings. Is it possible to work towards supporting this fully, or it would need changes to the Gradle integration of IDEA?

Reply:
Technically yes, this would be the state we want to achieve in the end.
A bit in the direction of https://github.com/detekt/detekt-intellij-plugin/issues/276.
The IntelliJ plugin should detect all config and baseline files automatically per module basis.
We do not communicate with the Gradle subsystem in IJ for now.
Idk if it would be possible to get the `DetektExtension` information from Gradle through IJ.

Reply:
Related https://github.com/detekt/detekt-intellij-plugin/issues/109